src/main/java/it/pagopa/transactions/utils/EcsEncoderLogMasker.java

```diff
@@ -1,28 +1,34 @@
 package it.pagopa.transactions.utils;
 
-import ch.qos.logback.classic.PatternLayout;
 import ch.qos.logback.classic.spi.ILoggingEvent;
+import co.elastic.logging.logback.EcsEncoder;
 
+import java.nio.charset.StandardCharsets;
 import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.List;
 import java.util.regex.Matcher;
 import java.util.regex.Pattern;
-import java.util.stream.Collectors;
 import java.util.stream.IntStream;
 
-public class LogMaskerPatternLayout extends PatternLayout {
+public class EcsEncoderLogMasker extends EcsEncoder {
     private Pattern patternString;
-    private List<String> maskPatterns = new ArrayList<>();
+    private final List<String> maskPatterns = new ArrayList<>();
+
+    public void setMaskPattern(String maskPatter) {
+        this.maskPatterns.add(maskPatter);
+    }
```

Comment:
Why use this over `addMaskPattern`?

Reply:
My bad, fixed with https://github.com/pagopa/pagopa-ecommerce-transactions-service/pull/255/commits/f1177f5f8dd109092af32f0b3fb3a2a19f2ea20f